src/main/kotlin/com/ajaxproject/telegrambot/bot/handlers/impl/TextEnteredHandler.kt

```diff
@@ -0,0 +1,43 @@
+package com.ajaxproject.telegrambot.bot.handlers.impl
+
+import com.ajaxproject.telegrambot.bot.enums.ConversationState
+import com.ajaxproject.telegrambot.bot.handlers.UserRequestHandler
+import com.ajaxproject.telegrambot.bot.model.UserRequest
+import com.ajaxproject.telegrambot.bot.model.UserSession
+import com.ajaxproject.telegrambot.bot.service.TelegramService
+import com.ajaxproject.telegrambot.bot.service.UserSessionService
+import com.ajaxproject.telegrambot.bot.utils.Id
+import com.ajaxproject.telegrambot.bot.utils.KeyboardUtils
+import com.ajaxproject.telegrambot.bot.utils.TextsUtils
+import org.springframework.stereotype.Component
+import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.KeyboardButton
+
+@Component
+class TextEnteredHandler(
+    val telegramService: TelegramService,
+    val text: TextsUtils,
+    val userSessionService: UserSessionService,
+) : UserRequestHandler() {
+    override fun isApplicable(request: UserRequest): Boolean {
+        return (isTextMessage(request.update)) &&
+            (ConversationState.WAITING_FOR_TEXT == request.userSession.state)
```

Comment:
```suggestion
        return isTextMessage(request.update) &&
            ConversationState.WAITING_FOR_TEXT == request.userSession.state
```

Reply:
Prefer using more cheep operation in boolean expression.

Comparing two enums definitely cheeper rather than `isTextMessage(request.update)`.
So, just swap the conditions:
```kotlin
return ConversationState.WAITING_FOR_TEXT == request.userSession.state
             && isTextMessage(request.update)
```